src/connectors/pb.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::errors::Result;
+use simd_json::StaticNode;
+use tremor_value::Value;
+
+pub(crate) fn maybe_string_to_pb(data: Option<&Value<'_>>) -> Result<String> {
+    if let Some(Value::String(s)) = data {
+        Ok(s.to_string())
+    } else if data.is_none() {
+        Ok("".to_string())
+    } else {
+        Err("Expected an json string to convert to pb string".into())
+    }
+}
+
+pub(crate) fn maybe_int_to_pbu64(data: Option<&Value<'_>>) -> Result<u64> {
+    if let Some(&Value::Static(StaticNode::U64(i))) = data {
```

Comment:
`as_u64` will cast the i64 if possible so you can simplify this

Reply:
Done - but `proptest` failure for `as_u64` case:

```
proptest: Aborting shrinking after the PROPTEST_MAX_SHRINK_ITERS environment variable or ProptestConfig.max_shrink_iters iterations (set 1024 to a large(r) value to shrink more; current configuration: 1024 iterations)
thread 'connectors::pb::test::prop_pb_u64_repeated' panicked at 'Test failed: not coercable to u64; minimal failing input: (vec, _index) = ([9270612287945480471], 0)
        successes: 1
        local rejects: 0
        global rejects: 0
', src/connectors/pb.rs:158:5
```

Reply:
I'll leave this method as is, the others work with the `as_*` variants and are easier to read 👍